src/arch/riscv/smp.c

```diff
@@ -20,7 +20,16 @@
 static struct sys_info * sys;
 
 static int
-dtb_parse_cpu (struct dtb_node * n) {
+configure_cpu (unsigned long fdt, int offset) {
+    off_t reg_addr = fdt_getreg_address(fdt, offset);
+    int lenp = 0;
+    char *status = fdt_getprop(fdt, offset, "status", &lenp);
+    int enabled = 1;
+    // sifive disables 1 CPU and indicates it with this property
+    if (status && !strcmp(status, "disabled")) {
```

Comment:
good catch, we ran into this problem before and it just spins forever on the QEMU platform, but doesn't on the real machine